### PR TITLE
Fix building with tbb when CUDA is not installed

### PR DESCRIPTION
**Pull request by cjmayo**
The default THRUST_DEVICE_SYSTEM is THRUST_DEVICE_SYSTEM_CUDA, causing the build to fail when looking for cuda_runtime_api.h.

Once this is set to THRUST_DEVICE_SYSTEM_TBB the build fails on linking.

----

Perhaps some people want to use CUDA for this??? In which case it looks like there is an implicit dependency here on CUDA and a missing option for those of us without NVIDIA hardware.

The device systems supported by Thrust are:

```c
#define THRUST_DEVICE_SYSTEM_CUDA    1
#define THRUST_DEVICE_SYSTEM_OMP     2
#define THRUST_DEVICE_SYSTEM_TBB     3
#define THRUST_DEVICE_SYSTEM_CPP     4
```
I've somewhat guessed here, probably don't want OpenMP as another dependency.

I'm using CUDA C++ Core Libraries (CCCL) 2.2.0. It contains Thrust's dependencies and the Thrust project is now archived.

### CODE REVIEW

**Comment by pca006132** [comment]
The CUDA dependency is just due to thrust being some Nvidia library and has a CUDA backend (that manifold do not use). I haven't checked the thrust version with CCCL yet, I guess it is a version issue... We will automatically pull thrust from the archived repo when thrust is not found on the machine.

**Comment by pca006132** [comment]
Btw, does it work with manifold alone? Not sure if it is something specific to openscad or an upstream issue.

**Comment by pca006132** [comment]
The linker issue on macos is weird. With manifold we are linking against either `TBB::tbb` or `${TBB_LINK_LIBRARIES}` if the target `TBB::tbb` is not defined. Not sure if that can fix the issue.

File: CMakeLists.txt [hunk 1]
```diff
@@ -857,7 +857,10 @@ if(EXPERIMENTAL AND ENABLE_TBB)
   target_compile_options(OpenSCAD PRIVATE 
     -DENABLE_TBB
     -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_TBB
+    -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_TBB
   )
+  pkg_check_modules(TBB tbb REQUIRED)
+  target_link_libraries(OpenSCAD PRIVATE ${TBB_LIBRARIES})
```
Comment (kintel) [inline CMakeLists.txt:863]: You should probably do this instead:
```
find_package(TBB REQUIRED)
target_link_libraries(OpenSCAD PRIVATE TBB::tbb)
```

It sounds like you want to patch Manifold, not OpenSCAD though.  Not sure how good it is to have both Manifold and OpenSCAD link TBB in their own way.

Comment (pca006132) [reply CMakeLists.txt:863]: I think this is not patching manifold: openscad does use thrust for parallel minkowski. As long as the tbb version is the same, I think linking tbb in both should work fine.

Comment (kintel) [reply CMakeLists.txt:863]: Oh, I hadn't realized; minkowski for Manifold does indeed use TBB.

**Review by pca006132 — APPROVED** [review]
LGTM

**Review by t-paul — APPROVED** [review]

**Pull request merged**